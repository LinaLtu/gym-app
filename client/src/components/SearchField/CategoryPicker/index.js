import CategoryPick from './CategoryPicker';

export default CategoryPick;