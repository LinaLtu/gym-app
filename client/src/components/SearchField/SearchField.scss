@import '../shared/common.scss';

.search-field {
    display: flex;
    align-items: flex-end;
    justify-content: space-around;
    align-items: center; 
    width: 100%;
    height: 70px;
    background-color:white;
    border-bottom: 2px solid #d4d4d4;

    &_label {
        font-weight: 600;
        font-size: 14px;
        margin-right: 10px;
    }

    &_category {
        display: flex;
        align-items: center;
    }

}