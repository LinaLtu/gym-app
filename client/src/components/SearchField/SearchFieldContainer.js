import React from 'react';
import SeachField from '/SeachField';

const SearchFieldContainer = () => {
    return (
        <SeachField />
    )
}


export default SearchFieldContainer;