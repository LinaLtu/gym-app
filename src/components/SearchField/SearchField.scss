.search-field {
    width: 100%;
    height: 70px;
    background-color:white;
    border-bottom: 2px solid #d4d4d4;

}