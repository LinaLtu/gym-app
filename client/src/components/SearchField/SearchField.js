import React, { useState } from 'react';
import { useQuery } from 'react-query';
import moment from 'moment';
import { getCourses } from '../../services/courseService'
import SearchDatePicker from './DatePicker';
import CategoryPicker from './CategoryPicker';
import './SearchField.scss';

const CSS_NAME = 'search-field'

const SearchField = ({ page }) => {

    const [category, setCategory] = useState([]);
    const [date, setDate] = useState(moment());

    const { status } = useQuery(['courses', page, date, category], (_, page, date, category) => getCourses(moment(date), category, page));

    if (status === 'loading') {
        return <div>Loading...</div>;
    }

    if (status === 'error') {
        return <div>Error!</div>;
    }

    return (
        <div className={CSS_NAME}>
            <div className={`${CSS_NAME}_start-date`}>
                <label for="byStartDate" className={`${CSS_NAME}_label`}>Search By Start Date </label>
                < SearchDatePicker setItemsByDate={setDate} />
            </div>
            <div className={`${CSS_NAME}_category`}>
                <label for="byStartDate" className={`${CSS_NAME}_label`}>Search By Category </label>
                < CategoryPicker setItemsByCategory={setCategory} />
            </div>
            <button className={`${CSS_NAME}_cta`}>Search!</button>
        </div>
    )
}

export default SearchField;