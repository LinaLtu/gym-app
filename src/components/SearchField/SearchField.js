import React from 'react';
import './SearchField.scss';

const CSS_NAME = 'search-field'

const SearchField = () => {
    return (
        <div className={CSS_NAME}>
            This is where you search
        </div>
    )
}

export default SearchField;