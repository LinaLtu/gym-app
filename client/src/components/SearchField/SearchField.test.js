import React from 'react';
import renderer from 'react-test-renderer';

import SearchField from './';

jest.mock('./DatePicker', () => 'SearchDatePicker');
jest.mock('./CategoryPicker', () => 'CategoryPicker');


describe('SearchField Component', () => {
    it('renders according to design', () => {
        const component = renderer.create(
            <SearchField />
        );
        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });
});