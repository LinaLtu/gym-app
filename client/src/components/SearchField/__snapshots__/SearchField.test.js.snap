// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchField Component renders according to design 1`] = `
<div
  className="search-field"
>
  <div
    className="search-field_start-date"
  >
    <label
      className="search-field_label"
      for="byStartDate"
    >
      Search By Start Date
    </label>
    <SearchDatePicker
      date={2020-08-03T17:18:50.582Z}
    />
  </div>
  <div
    className="search-field_category"
  >
    <label
      className="search-field_label"
      for="byStartDate"
    >
      Search By Category
    </label>
    <CategoryPicker />
  </div>
  <button
    className="search-field_cta"
  >
    Search!
  </button>
</div>
`;
